### PR TITLE
Raising error when tf.profiler failed to create logs

### PR DESCRIPTION
**Pull request by Elizaaaaa**
*Issue #, if available:*

In the 2.1.1 container, the tf profiler was not able to start when running as non-privileged mode. However, the testTensorBoard didn't through an error. 
By having this fix, the testTensorboard will exit as expected. 

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [ tensorflow] | [test]  | [ec2]

*Description:*

*Tests run:*
2020-06-25 20:30:05.895658: I tensorflow/core/profiler/lib/profiler_session.cc:225] Profiler session started.
2020-06-25 20:30:05.895711: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1259] Profiler found 8 GPUs
2020-06-25 20:30:05.896671: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcupti.so.10.1
2020-06-25 20:30:05.997330: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1307] function cupti_interface_->Subscribe( &subscriber_, (CUpti_CallbackFunc)ApiCallback, this)failed with error CUPTI_ERROR_INSUFFICIENT_PRIVILEGES
2020-06-25 20:30:06.005008: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1346] function cupti_interface_->ActivityRegisterCallbacks( AllocCuptiActivityBuffer, FreeCuptiActivityBuffer)failed with error CUPTI_ERROR_INSUFFICIENT_PRIVILEGES
Epoch 1/10
2020-06-25 20:30:06.316081: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2020-06-25 20:30:06.750155: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1329] function cupti_interface_->EnableCallback( 0 , subscriber_, CUPTI_CB_DOMAIN_DRIVER_API, cbid)failed with error CUPTI_ERROR_INVALID_PARAMETER
2020-06-25 20:30:06.750210: I tensorflow/core/profiler/internal/gpu/device_tracer.cc:88]  GpuTracer has collected 0 callback api events and 0 activity events.
1/1 [==============================] - 1s 756ms/sample - loss: 2.3157 - accuracy: 0.0000e+00
Epoch 2/10
1/1 [==============================] - 0s 7ms/sample - loss: 0.4891 - accuracy: 0.0000e+00
Epoch 3/10
1/1 [==============================] - 0s 7ms/sample - loss: 0.0818 - accuracy: 0.0000e+00
Epoch 4/10
1/1 [==============================] - 0s 7ms/sample - loss: 0.0182 - accuracy: 0.0000e+00
Epoch 5/10
1/1 [==============================] - 0s 6ms/sample - loss: 0.0054 - accuracy: 0.0000e+00
Epoch 6/10
1/1 [==============================] - 0s 6ms/sample - loss: 0.0020 - accuracy: 0.0000e+00
Epoch 7/10
1/1 [==============================] - 0s 7ms/sample - loss: 8.1839e-04 - accuracy: 0.0000e+00
Epoch 8/10
1/1 [==============================] - 0s 7ms/sample - loss: 3.9236e-04 - accuracy: 0.0000e+00
Epoch 9/10
1/1 [==============================] - 0s 6ms/sample - loss: 2.0395e-04 - accuracy: 0.0000e+00
Epoch 10/10
1/1 [==============================] - 0s 6ms/sample - loss: 1.1491e-04 - accuracy: 0.0000e+00
Checking the logs result: 
Cannot find Profiler logs!

*DLC image/dockerfile:*
669063966089.dkr.ecr.us-west-2.amazonaws.com/beta-tensorflow-training:2.1.1-gpu-py36-cu101-ubuntu18.04-2020-06-23-03-09-35

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/container_tests/bin/testTensorBoard [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 HOME_DIR=/test
 BIN_DIR=${HOME_DIR}/bin
-LOG_DIR=${HOME_DIR}/logs
+LOG_DIR=${HOME_DIR}/logs/train/plugins/profile
+
+[ -d $LOG_DIR ] && echo "Cleanning existing logs" && rm -rf $LOG_DIR
 
 python ${BIN_DIR}/testTensorBoard.py || exit 1
 
+LOG_RESULT=$(find $LOG_DUR -name "*.pb")
```
Comment (YYStreet) [inline test/dlc_tests/container_tests/bin/testTensorBoard:11]: Should it be $LOG_DIR?

Comment (YYStreet) [inline test/dlc_tests/container_tests/bin/testTensorBoard:11]: Should it be $LOG_DIR?

File: test/dlc_tests/container_tests/bin/testTensorBoard [hunk 2]
```diff
@@ -1,9 +1,16 @@
 #!/bin/bash
-
+  
 HOME_DIR=/test
 BIN_DIR=${HOME_DIR}/bin
-LOG_DIR=${HOME_DIR}/logs
+LOG_DIR=${HOME_DIR}/logs/train/plugins/profile
```
Comment (YYStreet) [inline test/dlc_tests/container_tests/bin/testTensorBoard:5]: Could it be ```/logs/train/plugins/profile```? I did some local tests and I might be wrong

**Comment by YYStreet** [comment]
Should the epochs not be set to 0? Currently it is 0 in testTensorBoard.py file

**Comment by TusharKanekiDey** [comment]
Is it expected that this test failed for TF1.15.3 as observed in the logs.

File: test/dlc_tests/container_tests/bin/testTensorBoard (hunk 2 above)
Comment (Elizaaaaa) [reply test/dlc_tests/container_tests/bin/testTensorBoard:5]: log dir should be {scirpt_path}/logs, not /logs.

**Comment by Elizaaaaa** [comment]
> Should the epochs not be set to 0? Currently it is 0 in testTensorBoard.py file

epoch > 0 is for visualizing the results in tensorboard ui. it will still generate logs for epochs=0

**Comment by Elizaaaaa** [comment]
> Is it expected that this test failed for TF1.15.3 as observed in the logs.
> If yes, should we make required changes to Tf1.15.3 Dockerfile.

shouldn't be. this issue is only observed in tf 2.1 for now. would you mind verify if running 15.3 in privileged mode will resolve the error? i didn't see error in 15.3 in logs.

**Comment by TusharKanekiDey** [comment]
> > Is it expected that this test failed for TF1.15.3 as observed in the logs.
> > If yes, should we make required changes to Tf1.15.3 Dockerfile.
> 
> shouldn't be. this issue is only observed in tf 2.1 for now. would you mind verify if running 15.3 in privileged mode will resolve the error? i didn't see error in 15.3 in logs.

Sure, I can test it out in privileged mode and see if it works.

**Comment by YYStreet** [comment]
> > Should the epochs not be set to 0? Currently it is 0 in testTensorBoard.py file
> 
> epoch > 0 is for visualizing the results in tensorboard ui. it will still generate logs for epochs=0

OK. I was looking through your PR description and there is training epochs > 0.

File: test/dlc_tests/container_tests/bin/testTensorBoard (hunk 2 above)
Comment (YYStreet) [reply test/dlc_tests/container_tests/bin/testTensorBoard:5]: The test script should be located in ```$BIN_DIR``` but not ```$HOME```.
Please also take a look of the test since it should pass on CPU images but failed.

I also tried to reproduce a successful run on functionality well images (e.g. 2.0.2 image or 2.2.0 CUDA10.1 image) and it seems to be ```/logs/``` but I still could not find a *.pb file.
The structure in my test is
```
/logs/train/plugins/profile/2020-06-26_19-35-07/local.trace
```

**Comment by YYStreet** [comment]
Can we separate the TF1 part in another PR? We can skip for the TF1 in this ticket and try to merge this one in.
